casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -215,29 +214,61 @@ object ProtoUtil {
 
   def unsignedBlockProto(body: Body,
                          header: Header,
-                         justifications: Seq[Justification]): BlockMessage =
+                         justifications: Seq[Justification]): BlockMessage = {
+    val hash = hashUnsignedBlock(header, justifications)
+
     BlockMessage()
-      .withBlockHash(protoHash(header))
+      .withBlockHash(hash)
       .withHeader(header)
       .withBody(body)
       .withJustifications(justifications)
+  }
+
+  def int32ToByteArray(i: Int): Array[Byte] = {
```

Comment:
I'm a little worried about this because our block hashes will now be tied to the way the JVM represents integers, whereas previously everything was based on protobuf serialization. Perhaps it is worth considering making a `BlockHashData` protobuf with fields for everything that needs to be included in the hash, then simply calling hashing the result of `toByteArray`.